openghg/store/base/_base.py

```diff
@@ -233,26 +237,31 @@ def _standardise_from_file(
             filters=filters,
         )
 
+        if filepath is not None:
+            filepaths = normalise_to_filepath_list(filepath)
+        else:
+            filepaths = []
         for x in datasource_uuids:
-            if isinstance(filepath, list) and len(filepath) == 1:
-                filepath = filepath[0]
-
-            if isinstance(filepath, Path):
-                x.update({"file": filepath.name})
-                logger.info(f"Completed processing: {filepath.name}.")
-            elif isinstance(filepath, list):
-                filepath_str = ", ".join([fp.name for fp in filepath])
+            if not filepaths:
+                continue
+            if len(filepaths) == 1:
+                fp = filepaths[0]
+                x.update({"file": fp.name})
+                logger.info(f"Completed processing: {fp.name}.")
+            else:
+                filepath_str = ", ".join(fp.name for fp in filepaths)
                 x.update({"files": filepath_str})
                 logger.info(f"Completed processing files: {filepath_str}.")
 
         return datasource_uuids
 
     def read_file(
```

Comment:
Would likely want to rename this as well - `read_data` is a suggestion but if there is already a function named this we may need to rename the other function to something else.